Framework/DataHandling/src/ImggAggregateWavelengths.cpp

```diff
@@ -0,0 +1,834 @@
+                                               size_t bands) {
+
+  Poco::Path path(inputPath);
+  auto inputSubDirs = findInputSubdirs(path);
+
+  if (inputSubDirs.empty()) {
+    g_log.warning() << "Could not find any input files or directories in "
+                    << inputPath
+                    << ". Nothing will be produced in the output path."
+                    << std::endl;
+  }
+
+  size_t count = 0;
+  for (const auto &subdir : inputSubDirs) {
+    processDirectory(subdir, bands, outputPath, outPrefixBands, count++);
+  }
+}
+
+void ImggAggregateWavelengths::aggIndexBands(const std::string &inputPath,
+                                             const std::string &outputPath,
+                                             const std::string &rangesSpec) {
+
+  auto ranges = rangesFromStringProperty(rangesSpec, PROP_INDEX_RANGES);
+
+  Poco::Path path(inputPath);
+  auto inputSubDirs = findInputSubdirs(path);
+
+  if (inputSubDirs.empty()) {
+    g_log.warning() << "Could not find any input files or directories in "
+                    << inputPath
+                    << ". Nothing will be produced in the output path."
+                    << std::endl;
+  }
+
+  size_t count = 0;
+  for (const auto &subdir : inputSubDirs) {
+    processDirectory(subdir, ranges, outputPath, outPrefixBands, count++);
+  }
+}
+
+void ImggAggregateWavelengths::aggToFBands(const std::string & /*inputPath*/,
+                                           const std::string & /*outputPath*/,
+                                           const std::string & /*ranges*/) {
+
+  throw std::runtime_error("The property " + PROP_TOF_RANGES +
+                           "is not supported at the moment");
+}
+
+std::vector<std::pair<size_t, size_t>>
+ImggAggregateWavelengths::rangesFromStringProperty(
+    const std::string &rangesSpec, const std::string &propName) {
+  Mantid::Kernel::StringTokenizer rangeTokens(
+      rangesSpec, ",", Mantid::Kernel::StringTokenizer::TOK_IGNORE_EMPTY |
+                           Mantid::Kernel::StringTokenizer::TOK_TRIM);
+
+  if (rangeTokens.count() <= 0) {
+    throw std::invalid_argument(
+        "Could not find any valid range of values in the property " + propName);
+  }
+
+  std::vector<std::pair<size_t, size_t>> ranges;
+  for (const auto &str : rangeTokens) {
+    const std::string sep = "-";
+    Mantid::Kernel::StringTokenizer minMaxTokens(
+        str, sep, Mantid::Kernel::StringTokenizer::TOK_IGNORE_EMPTY |
+                      Mantid::Kernel::StringTokenizer::TOK_TRIM);
+    if (2 != minMaxTokens.count()) {
+      throw std::invalid_argument(
+          "Could not parse a minimum and maximum value separated by '" + sep +
+          "' from the string: " + str);
+    }
+
+    try {
+      ranges.emplace_back(
+          std::make_pair(boost::lexical_cast<size_t>(minMaxTokens[0]),
+                         boost::lexical_cast<size_t>(minMaxTokens[1])));
```

Comment:
With `emplace_back` you don't need to use `std::make_pair`. See, for example, #15874.